src/select/Select.ts

```diff
@@ -0,0 +1,419 @@
+import { WidgetBase, onPropertiesChanged } from '@dojo/widget-core/WidgetBase';
+import { DNode, PropertiesChangeEvent } from '@dojo/widget-core/interfaces';
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
+import WidgetRegistry from '@dojo/widget-core/WidgetRegistry';
+import { v, w } from '@dojo/widget-core/d';
+import uuid from '@dojo/core/uuid';
+import { assign } from '@dojo/core/lang';
+import { find, includes } from '@dojo/shim/array';
+import Label, { LabelOptions } from '../label/Label';
+import SelectOption, { OptionData } from './SelectOption';
+import * as css from './styles/select.m.css';
+
+/**
+ * @type SelectProperties
+ *
+ * Properties that can be set on a Select component
+ *
+ * @property customOption   Custom widget constructor for options. Should use SelectOption as a base
+ * @property describedBy    ID of an element that provides more descriptive text
+ * @property disabled       Prevents the user from interacting with the form field
+ * @property formId         ID of a form element associated with the form field
+ * @property invalid        Indicates the value entered in the form field is invalid
+ * @property label          Label settings for form label text, position, and visibility
+ * @property multiple       Whether the widget supports multiple selection
+ * @property name           The form widget's name
+ * @property options        Array of data for the select options' value, text content, and state
+ * @property readOnly       Allows or prevents user interaction
+ * @property required       Whether or not a value is required
+ * @property useNatveSelect Use the native <select> element if true
+ * @property value          The current value
+ * @property onBlur         Called when the input loses focus
+ * @property onChange       Called when the node's 'change' event is fired
+ * @property onClick        Called when the input is clicked
+ * @property onFocus        Called when the input is focused
+ * @property onKeyDown      Called on the input's keydown event
+ */
+export interface SelectProperties extends ThemeableProperties {
+	customOption?: any;
+	describedBy?: string;
+	disabled?: boolean;
+	formId?: string;
+	invalid?: boolean;
+	label?: string | LabelOptions;
+	multiple?: boolean;
+	name?: string;
+	options?: OptionData[];
+	readOnly?: boolean;
+	required?: boolean;
+	useNativeSelect?: boolean;
+	value?: string;
+	onBlur?(event: FocusEvent): void;
+	onChange?(option: OptionData): void;
+	onClick?(event: MouseEvent): void;
+	onFocus?(event: FocusEvent): void;
+	onKeyDown?(event: KeyboardEvent): void;
+}
+
+// This should have a lookup somewhere
+const keys = {
```

Comment:
Probably. I think it might be better to do it in a separate PR, though, and change it in ComboBox, Tabs, and wherever else as well.